components/formFields.js

```diff
@@ -38,34 +38,82 @@ export const CheckBoxFormField = (props) => {
   );
 };
 
-export const CustomCheckBoxContainerParent = (props) => {
+export const LabelListContainer = (props) => {
   return (
-    <View style={styles.customCheckBoxContainerParent}>
+    <View style={styles.customCheckBoxContainer}>
       <TextSourceSans style={styles.formLabel}>{props.field}</TextSourceSans>
       {props.children}
       {props.children.touched && <TextSourceSans style={styles.formError}>{props.error}</TextSourceSans>}
     </View>
   );
 };
 
-export const CustomCheckBoxFormField = (props) => {
-  const plusIcon = (
-    <IconFA name='plus' size={16} color={COLORS.text.main} />
-  );
+export const LabelList = props => {
+  const [ selectedLabels, setSelectedLabels ] = useState(props.selectedLabels);
+
+  const handleItemPressed = selectedLabel => {
+    let selectedLabelsCopy = [...selectedLabels];
+    /* the following code might be hard to read. It is comparing if the newly
+     selected label was selected before or not. If yes, then it removes the label,
+     if not it adds to the list. Then updating the state and sending to its parent
+     Note: this has to be done separately because setting State does not immediately change
+     the actual state (FIXME). */
+    const labelIndex = selectedLabels.findIndex(sl => sl.id === selectedLabel.id);
+    (labelIndex === -1) && (selectedLabelsCopy = [...selectedLabels, selectedLabel]);
+    (labelIndex !== -1) && [...selectedLabelsCopy.splice(labelIndex, 1)];
+    setSelectedLabels([...selectedLabelsCopy]);
+    props.onIconPress(selectedLabelsCopy);
+  };
+
+  const getCheckState = (choice) => {
+    return !!props.selectedLabels.find(sl => sl.id === choice.id);
+  };
+
+  return props.labelChoices.map(choice => (
+    <LabelFormField
+      key={`customCheckBox_${choice.value}`}
```

Comment:
this doesn't seem to work getting `Encountered two children with the same key, `customCheckBox_undefined`.` error